shopinvader/models/product_product.py

```diff
@@ -59,12 +62,26 @@ def _add_to_cart_allowed(self, backend, partner=None):
             self.shopinvader_bind_ids.filtered(lambda x: x.backend_id == backend)
         )
 
-    def _get_invader_variant(self, backend, lang):
+    def _get_invader_variant(self, backend, lang, safe_default = False):
         """Retrieve invader variant by backend and language.
 
         :param backend: backend recordset
         :param lang: lang code
+        :param safe_default: true to fallback to backend lang 
         """
+        variant = self._find_variant(backend, lang)
+        if not variant and safe_default:
+            lang = self.shopinvader_bind_ids.filtered(
+                lambda x: x.backend_id == backend).lang_id[0].code
```

Comment:
```suggestion
            lang = self._find_invader_variant(backend, backend.lang_ids[0].code)
```

Reply:
are you sure about backend.lang_id?

Reply:
please add a test to cover this case

Reply:
I would use `fields.first(backend.lang_ids)` instead of `backend.lang_ids[0]`

Reply:
It's actually lang_id that we need to use (no need for indexing) as lang_id stores the default language of the backend.

Reply:
bind_ids is a m2o, which means that `bind_ids.backend_id.lang_id` will return multiple langs.

Reply:
I will add fields.first there to be on the safe side, but can there be more than 1 binded id per backend after applying the filter anyway?

Reply:
You can have multiple binds referenced by self. Each one of those might have a different backend, and those backends might have different langs.

Reply:
But here we're talking about the lang of the bind_id and not the lang of the backend, right? I was just wondering if there was any case where we could find more than 1 shopinvader_bind_ids per backend for a product variant.

In any case, I've added the fields.first() to make sure we never get more than 1 shopinvader_bind_ids, this will only give 1 language every time.

Reply:
> But here we're talking about the lang of the bind_id and not the lang of the backend, right? I was just wondering if there was any case where we could find more than 1 shopinvader_bind_ids per backend for a product variant.

That's the point. Here you can have N bindings for the same backend for N languages.

Reply:
And each of those bindings will have 1 lang_id so fields.first should only be applied to shopinvader_bind_ids, and not shopinvader_bind_ids.lang_id, correct?

Reply:
In order to not retrieve `bakend_id and then `lang_id` for each one of the bindings, better use `fields.first(self.shopinvader_bind_ids)`, I think.

Reply:
Current code: 
`lang = fields.first(self.shopinvader_bind_ids.filtered(lambda x: x.backend_id == backend)).lang_id.code`


If I use `fields.first(self.shopinvader_bind_ids)` instead, I could potentially get a bind from a different backend, which wouldn't be correct.
`lang_id` is only going to be accessed for the filtered `shopinvader_bind_id` record.

Reply:
Your current code is the right one. Sorry to take shortcuts in my comments :pray: